### ISSUE TITLE
How to work with this on web installation

### ISSUE THREAD

**Issue by siddarth9** [searx/searx#1024]
How to install this searx on web host

**Comment by dalf** [comment]
is that the link you are looking for : https://asciimoo.github.io/searx/dev/install/installation.html#installation  ?

**Comment by siddarth9** [comment]
What was the secret key where do I can get that secret key to add in setting.yml

**Comment by dalf** [comment]
It can be any value you want, it's a kind of password. 

Searx can be use to proxy images. Have a look to the source of this page : 
https://searx.me/?q=%21goi+searx&language=all&time_range=&safesearch=0&theme=simple&image_proxy=True

The images are not fetched from google or whatever, but only from searx : the purpose is to protected your privacy.
You can see that there is a hash parameter for each images : the value is calculated using the secret key. Try to change it : you won't get the image. Without this hash, anyone would be able to fetch images from anywhere.

So to sum up, just do 
```sh
sed -i -e "s/ultrasecretkey/`openssl rand -hex 16`/g" searx/settings.yml
```
and it will be fine.

**Comment by siddarth9** [comment]
I'm interested in using on my own domain so I want it to use on live how can I install this on cpanel of my web host
Please help me find solution

**Comment by siddarth9** [comment]
Is there anyone who could help with this

**Comment by misnyo** [comment]
@siddarth9 it depends on your webhosting, but generally it is advised to install searx on a vps with shell access. Does your webhosting provide that?

**Comment by siddarth9** [comment]
My webhost is cpanel shared  hosting On vps have root access my webhost proveiders ehost.com 


.please help me find one solution for this

**Comment by misnyo** [comment]
I'm not sure about if searx can be installed on cpanel, but I would go for the vps root access, and follow the instructions on the link provided by @dalf ;)
